aten/src/ATen/hip/impl/HIPGuardImplMasqueradingAsCUDA.cpp

```diff
@@ -12,3 +12,4 @@
 // This hack can be removed once PyTorch is out-of-place HIPified, and
 // doesn't pretend CUDA is HIP.
 C10_REGISTER_GUARD_IMPL(CUDA, at::cuda::HIPGuardImplMasqueradingAsCUDA);
+C10_REGISTER_GUARD_IMPL(HIP, at::cuda::HIPGuardImplMasqueradingAsCUDA);
```

Comment:
This looks wrong. Even in Torch, if I access something referring to it by its true HIP name, I had better get something that actually reports itself to be HIP, and not lying to say it's CUDA.

Reply:
This was an attempt to solve a rocm CI issue, but it doesn't seem to have helped. cc @bddppq